### ISSUE TITLE
Create a mixin for "updatable" visualization component

### ISSUE THREAD

**Issue by ronichoudhury** [Kitware/candela#258]
This would mainly consist of an `update(options)` method.

The semantics of the method could vary (i.e. we won't, and really can't, enforce them generally) but generally speaking, any visualization that doesn't have an `update()` method could gain one by taking all the vis prep done in the constructor, moving it into `update()`, then calling `this.update(options)` in the constructor in lieu of doing the prep directly. If updating a visualization should cause an immediate rerender, then `update()` can call `render()` directly. Other options would be to emit a signal or to do nothing.

@alex-r-bigelow, what do you think?

**Comment by ronichoudhury** [comment]
I could also be convinced that this should just be part of `VisComponent`, perhaps with a default empty implementation (i.e., not one that throws an error, etc.).

**Comment by alex-r-bigelow** [comment]
Yeah, we're already doing the wipe-and-re-render approach as the default behavior in Resonant Lab if an `update()` function doesn't exist—moving that logic into `VisComponent` makes a lot of sense to me.

As an aside, an `update()` feature—especially allowing a view to update **without** a full re-render—is absolutely critical if we want to enable any kind of Hanpuku-style vis customization. So my vote would be to keep it simple, where the default is to wipe and re-render if no `update()` feature exists, but let the component take over whether and how to call `render()` if it implements `update()`
